### PR TITLE
[Communication] - #2bfg9g4 - [Centralize Notification] Create Internal Package Provider for Wavecell

### PR DESCRIPTION
**Pull request by fairyhunter13**
# Changelog

- Removing unused requests, responses, and clients :fire:
- Refactoring client :recycle:
- Conforming the client to the new OAS specification of wavecell.com :alien:
- Using iso8601.Time type :sparkles:
- Refactor code a little :recycle:
- Completing tests :white_check_mark:

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -2,77 +2,146 @@ package wavecell
...
 		return
 	}
-	path := fmt.Sprintf(SINGLE_SMS_ENDPOINT, s.Config.ClientID)
-	r, err = s.defaultRequest(b, path)
+
+	c = (new(client)).Assign(o)
+	return
+}
+
+// ResponseSendSMS is the response struct for SendSMSV1.
```
Comment (asqolani-flip) [inline client.go:48]: mas nanti struct v2 apakah beda?

**Review by asqolani-flip — APPROVED** [review]
approve with comment

File: client.go (hunk 1 above)
Comment (fairyhunter13) [reply client.go:48]: @asqolani-flip Ya, bisa jadi mas, Wavecell menggunakan **API versioning** di endpoint untuk mengirim SMS. Kemungkinan jika ada V2, maka struktur request dan response bisa berubah juga.
image

Comment (asqolani-flip) [reply client.go:48]: oh sorry berati v1 di third party ya mksdnya, saya kira di send message notification service. noted mas thanks

**Pull request merged**